src/java.desktop/share/classes/javax/swing/plaf/basic/BasicFileChooserUI.java

```diff
@@ -706,17 +709,19 @@ public void valueChanged(ListSelectionEvent evt) {
                         if (objects.length == 1
                             && ((File)objects[0]).isDirectory()
                             && chooser.isTraversable(((File)objects[0]))
-                            && (useSetDirectory || !fsv.isFileSystem(((File)objects[0])))) {
+                            && (useSetDirectory || (!fsv.isFileSystem(((File)objects[0]))
+                            && !Files.isSymbolicLink(Paths.get(((File)objects[0]).getPath()))))) {
                             setDirectorySelected(true);
                             setDirectory(((File)objects[0]));
                         } else {
                             ArrayList<File> fList = new ArrayList<File>(objects.length);
                             for (Object object : objects) {
                                 File f = (File) object;
                                 boolean isDir = f.isDirectory();
+                                Path path = Paths.get(f.getPath());
```

Comment:
```suggestion
                                Path path = f.toPath();
```

You can directly convert a `File` to `Path`. Please update this in all the places.